packages/babel-generator/src/buffer.js

```diff
@@ -14,6 +14,7 @@ export default class Buffer {
     this._indent = format.indent.base;
     this.format = format;
     this.buf = "";
+    this.last = "";
```

Comment:
See below, every time we add to the buffer we simply get the `last` from the string argument as opposed to accessing the buffer.